pkg/apis/core/v1beta1/types_seed.go

```diff
@@ -270,8 +270,24 @@ type SeedSettings struct {
 
 // SeedSettingExcessCapacityReservation controls the excess capacity reservation for shoot control planes in the seed.
 type SeedSettingExcessCapacityReservation struct {
-	// Enabled controls whether the excess capacity reservation should be enabled.
-	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
+	// Enabled controls whether the default excess capacity reservation should be enabled. When not specified, the functionality is enabled.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty" protobuf:"bytes,1,opt,name=enabled"`
+	// Configs configures excess capacity reservation deployments for shoot control planes in the seed.
+	// +optional
+	Configs []SeedSettingExcessCapacityReservationConfig `json:"configs,omitempty" protobuf:"bytes,2,rep,name=configs"`
+}
+
+// SeedSettingExcessCapacityReservationConfig configures excess capacity reservation deployments for shoot control planes in the seed.
+type SeedSettingExcessCapacityReservationConfig struct {
+	// Resources specify the resource requests and limits of the excess-capacity-reservation pod.
+	Resources corev1.ResourceList `json:"resources" protobuf:"bytes,1,rep,name=resources,casttype=ResourceList,castkey=ResourceName"`
```

Comment:
When I remove the protobuf part from this type and regenerate it, I get 
```diff
diff --git a/pkg/apis/core/v1beta1/types_seed.go b/pkg/apis/core/v1beta1/types_seed.go
index 2bbed7f24d..34a5e10aa3 100644
--- a/pkg/apis/core/v1beta1/types_seed.go
+++ b/pkg/apis/core/v1beta1/types_seed.go
@@ -281,7 +281,7 @@ type SeedSettingExcessCapacityReservation struct {
 // SeedSettingExcessCapacityReservationConfig configures excess capacity reservation deployments for shoot control planes in the seed.
 type SeedSettingExcessCapacityReservationConfig struct {
 	// Resources specify the resource requests and limits of the excess-capacity-reservation pod.
-	Resources corev1.ResourceList `json:"resources" protobuf:"bytes,1,rep,name=resources,casttype=ResourceList,castkey=ResourceName"`
+	Resources corev1.ResourceList `json:"resources" protobuf:"bytes,1,rep,name=resources,casttype=k8s.io/api/core/v1.ResourceList,castkey=k8s.io/api/core/v1.ResourceName"`
 	// NodeSelector specifies the node where the excess-capacity-reservation pod should run.
 	// +optional
 	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty" protobuf:"bytes,2,opt,name=nodeSelector"`
```
Can you please check?

Reply:
Maybe we can also update https://github.com/gardener/gardener/blob/master/docs/development/changing-the-api.md#changing-the-api-1 to not copy protobuf tags from other places and always generate it.

Reply:
good hint thanks 👍

Reply:
I just fixed the protobuf tag and updated the docs too.